cpp/src/common/database.cpp

```diff
@@ -0,0 +1,198 @@
+#include <chrono>
```

Comment:
Just observation, it looks like something like serialization code from this file has to be part of the driver